### DISCUSSION TITLE
Artifacts and Runs not getting recorded

### DISCUSSION THREAD

**Discussion by debanshu-mutt** [mlflow/mlflow#7090] [Q&A]
Hello, I am running an MLFlow Server on a Azure VM using mlflow server command :
mlflow server --backend-store-uri sqlite:///mlflow.db --default-artifact-root wasbs://mlflowartifacts@storageaccountname.blob.core.windows.net/mlflowartifacts --host 0.0.0.0 --port 5000

Using SSH I have connected my local system to that VM and in my local I have set the mlflow tracking uri to public ip : 5000 of the VM
But the problem is in the mlflow ui the experiments and the runs are visible 
but no folders for the runs or experiments are getting listed in the VM directory 
The code is not breaking anywhere, yet even artifacts are not loading to Blob.

wasbs://mlflowartifacts@storageaccountname.blob.core.windows.net/1/8f7b9f57b07c4e068dee508b44f04938/artifacts
this path is getting shown but in azure nothing is visible 
Can someone assist with this?

**Comment by JHibbard** [comment]
Hi debanshu-mutt,

It sounds like you might be logging artifacts locally, where your MLflow code is executing, but not pushing to your Azure storage. Are artifacts being logged on the local system you're executing your logging logic from?

Thanks!
Jim

**Reply by debanshu-mutt** [reply]
Hi Jim,
I just checked they are getting logged on the local system in the folder where the code is.
But my tracking_uri points to the and my artifacts_uri is pointing to the Azure Blob Storage
I am printing both in the code and getting these values
Then why are the artifacts not getting logged?
Moreover the mlruns and experiments folder how to make sure when we run the code in local, it gets created in the VM?

**Reply by JHibbard** [reply]
Hi debanshu-mutt,

When using Azure Blob Storage for artifacts, you'll have to configure some additional environment variables described [here](https://www.mlflow.org/docs/latest/tracking.html#azure-blob-storage). Also, you'll have to set all of these environment variables both in the process running your MLflow tracking server **and** the process running your client code:

MLflow Server:
```sh
export AZURE_STORAGE_CONNECTION_STRING=...
mlflow server \
    --backend-store-uri sqlite:///mlflow.db \
    --default-artifact-root wasbs://<container>@<storage-account>.blob.core.windows.net/<path> \ 
    --host 0.0.0.0 \
    --port 5000
```

Running Client Code (assuming it's in `training.py` file):
```sh
export AZURE_STORAGE_CONNECTION_STRING=...
export MLFLOW_TRACKING_URI=sqlite:///mlflow.db
python training.py
```

MLflow has several possible deployment scenarios worth looking through [here](https://www.mlflow.org/docs/latest/tracking.html#how-runs-and-artifacts-are-recorded). Let me know if this solves your issue, we can definitely iterate until we have you going!

Thanks,
Jim
